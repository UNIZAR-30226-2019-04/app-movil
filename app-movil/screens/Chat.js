import React, { Component } from "react";
import { GiftedChat } from "react-native-gifted-chat";
import { API_BASE } from "../config";
import axios from "axios";
import SocketIOClient from "socket.io-client/dist/socket.io";
import io from "socket.io-client/dist/socket.io";

export default class Settings extends Component {
  constructor(props) {
    super(props);
    // Creating the socket-client instance will automatically connect to the server.
    this.socket = SocketIOClient(`${API_BASE}/mychat`, {
      transports: ["websocket"]
    });
  }
  state = {
    messages: [],

    input: "",
    user: "",
    room: "",
    receiver: "",
    token: "",
    conversaciones: []
  };

  static navigationOptions = ({ navigation }) => ({
    title: navigation.getParam("receiver"),
    headerTitleStyle: { textAlign: "center", alignSelf: "center" },
    headerStyle: {
      backgroundColor: "white"
    }
  });

  parseMsg(data) {
    const user = this.props.navigation.state.params.user;

    var res = {
      _id: data.id,
      text: data.text,
      createdAt: data.created_date,
      user: {
        _id: data.user === user ? 1 : 2,
        name: data.user === user ? "Me" : "Person",
        avatar: "https://placeimg.com/140/140/any"
      }
    };
    //this.feed.push(res);
    console.log(res);
    return res;
  }

  fetchMessages = (room, token) => {
    console.log("Fetch", room, token);
    const URL = `${API_BASE}/chat/` + room;
    axios
      .get(URL, {
        headers: {
          "Content-Type": "application/json",
          Authorization: token
        }
      })
      .then(response => {
        const msg = response.data.data;
        const messages = msg.map(m => this.parseMsg(m));
        this.setState({ messages: messages.reverse() });
      });
  };

  componentWillMount() {
    const { navigation } = this.props;
    const user = this.props.navigation.state.params.user;
    const token = navigation.state.params.token;

    const room = navigation.state.params.room;
    const receiver = navigation.state.params.receiver;
    console.log("Variables", user, room, receiver, token);
    this.setState({ user: user, room: room, receiver: receiver, token: token });

    this.fetchMessages(room, token);

    console.log("Params", this.props.navigation.state.params);
    this.socket.emit("JOINED", {
      room: navigation.getParam("room")
    });

    this.socket.on("MESSAGE", data => {
      console.log("MESSAGE", data);
      var msg = this.parseMsg(data);
      this.setState(previousState => ({
        messages: GiftedChat.append(previousState.messages, msg)
      }));
    });
  }

  onSend(messages = []) {
    const { navigation } = this.props;

    const msg = messages[0];
    console.log("SEND_MESSAGE", messages[0]);

    this.socket.emit("SEND_MESSAGE", {
      user: navigation.getParam("user"),
      text: msg.text,
      conversacion: navigation.getParam("room"),
      room: navigation.getParam("room"),
      created_date: msg.createdAt
    });

    /*     this.setState(previousState => ({
      messages: GiftedChat.append(previousState.messages, messages)
    })); */
  }

  render() {
    return (
      <GiftedChat
        inverted={true}
        messages={this.state.messages}
        onSend={messages => this.onSend(messages)}
        user={{
          _id: 1
        }}
      />
    );
  }
}
