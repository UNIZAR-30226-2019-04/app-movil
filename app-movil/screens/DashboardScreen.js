import React, { Component } from "react";
import { View, Text, StyleSheet, Button, Field } from "react-native";
import Icon from "@expo/vector-icons/Ionicons";
import { SearchBar } from "react-native-elements";


export default class DashboardScreen extends Component {
  render() {
    return (
      <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
        <Text>DashboardScreen</Text>
      </View>
    );
  }
}
