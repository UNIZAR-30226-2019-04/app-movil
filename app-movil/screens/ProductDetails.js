import React, { Component } from "react";
import { View, Text, StyleSheet, Image, Dimensions } from "react-native";
import { AsyncStorage } from "react-native";
import { Rating, AirbnbRating, Button } from "react-native-elements";
import Icon from "react-native-vector-icons/FontAwesome";
import ModalDropdown from "react-native-modal-dropdown";

const width = Dimensions.get("window").width;
let _this = null;

export default class ProductDetails extends Component {
  state = {
    search: "",
    modalVisible: null,
    product: "",
    room: 1,
    user: "unzurdo@gmail.com",
    receiver: "alberto@gmail.com",
    token: ""
  };

  static navigationOptions = ({ navigation }) => {
    return {
      // Your custom header

      headerRight: (
        <View
          style={{
            flex: 1,
            marginLeft: 10,
            flexDirection: "row",
            marginRight: 10
          }}
        >
          <View style={{ marginRight: 15 }}>
            <Icon
              style={{ marginTop: 6, marginHorizontal: 5 }}
              name="heart"
              size={30}
              color="white"
            />
          </View>

          <View style={{ marginRight: 15 }}>
            <Icon
              style={{ marginTop: 6, marginHorizontal: 5 }}
              name="share-alt-square"
              size={32}
              color="white"
            />
          </View>

          <ModalDropdown
            options={["Reportar anuncio"]}
            dropdownStyle={{ marginTop: -10, height: 40 }}
            renderRow={(a, b, c) => _this._renderRow(a, b, c)}
          >
            <Icon
              style={{ marginTop: 6, marginHorizontal: 5 }}
              name="ellipsis-v"
              size={35}
              color="white"
            />
          </ModalDropdown>
        </View>
      )
    };
  };

  _method = () => {
    this.props.navigation.navigate("Chat");
  };

  _renderRow(rowData, rowID, highlighted) {
    console.log(rowData, rowID, highlighted);
    return (
      <Button
        title={rowData}
        type="clear"
        //onPress={() => this.props.navigation.navigate("Welcome")}
        style={{ color: "black", width: 200 }}
      />
    );
  }
  componentDidMount() {
    _this = this;

    const { setParams } = this.props.navigation;
    const { state } = this.props.navigation;
    const product = this.props.navigation.state.params.product;
    console.log(product);
    this.setState({ product: product });
  }

  render() {
    console.log("render", this.state.product);
    const width = Dimensions.get("window").width;
    return (
      <View style={{ flex: 1 }}>
        <View style={styles.container}>
          <View style={{ flex: 1 }}>
            <Image
              source={{ uri: this.state.product.url }}
              style={{
                resizeMode: "cover",
                width: width,
                height: 200,
                borderRadius: 12
              }}
            />
          </View>
          <View style={{ flex: 2 }}>
            <View>
              <View style={styles.lineStyle} />

              <Text
                style={{
                  fontSize: 20,
                  fontWeight: "500",
                  paddingHorizontal: 10
                }}
              >
                {this.state.product.price}â‚¬
              </Text>
            </View>

            <View style={{ marginBottom: 5 }}>
              <Text style={{ paddingHorizontal: 10, color: "grey" }}>
                {this.state.product.name}
              </Text>
            </View>

            <View style={{ marginBottom: 5 }}>
              <Text
                //ellipsizeMode="tail"
                //numberOfLines={6}
                style={{ paddingHorizontal: 10, color: "grey" }}
              >
                {this.state.product.description}
              </Text>
            </View>
            <View style={styles.lineStyle} />

            <View style={{ marginBottom: 5 }}>
              <Text style={{ paddingHorizontal: 10, marginBottom: 120 }}>
                USUARIO
              </Text>
            </View>
          </View>
        </View>
        <View
          style={{
            width: width,
            position: "absolute",
            bottom: 0
          }}
        >
          <Button
            title="Chat"
            onPress={() =>
              this.props.navigation.navigate("ChatTabNavigator", {
                room: this.state.room,
                user: this.state.user,
                receiver: this.state.receiver,
                token: this.state.token
              })
            }
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 10
  },
  button: {
    marginHorizontal: 2,
    width: width / 2 - 13,
    height: 42,
    borderRadius: 6,
    borderWidth: 2,
    borderColor: "#fff"
  },
  lineStyle: {
    borderWidth: 1,
    borderColor: "#eeee",
    margin: 4
  }
});
