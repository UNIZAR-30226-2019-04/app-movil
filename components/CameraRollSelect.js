import React from "react";
import {
  StyleSheet,
  Text,
  View,
  Button,
  ScrollView,
  Image
} from "react-native";
import ImageBrowser from "./ImageBrowser";

export default class CameraRollSelect extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      imageBrowserOpen: false,
      photos: []
    };
  }
  imageBrowserCallback = callback => {
    callback
      .then(photos => {
        console.log(photos);
        this.setState({
          imageBrowserOpen: false,
          photos
        });
        this.props.saveImages(photos);
      })
      .catch(e => console.log(e));
  };

  renderImage(item, i) {
    return (
      <Image
        style={{ height: 100, width: 100, marginHorizontal: 2 }}
        source={{ uri: item.file }}
        key={i}
      />
    );
  }
  render() {
    if (this.state.imageBrowserOpen) {
      return <ImageBrowser max={4} callback={this.imageBrowserCallback} />;
    }
    return (
      <View style={styles.container}>
        <Button
          title="Seleccionar ImÃ¡genes"
          onPress={() => this.setState({ imageBrowserOpen: true })}
        />

        <ScrollView
          style={{ flex: 1, flexDirection: "row" }}
          horizontal={true}
          showsHorizontalScrollIndicator={false}
        >
          {this.state.photos.map((item, i) => this.renderImage(item, i))}
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 5,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center"
  }
});
