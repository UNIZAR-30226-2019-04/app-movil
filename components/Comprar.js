import React from "react";
import { View, Text, StyleSheet, Dimensions, TouchableOpacity, Modal, WebView } from "react-native";

import {
    RkText,
    RkTextInput,
    RkAvoidKeyboard,
    RkTheme,
    RkStyleSheet
  } from "react-native-ui-kitten";
  import { API_BASE } from "../config";
  import axios from "axios";
const width = Dimensions.get("window").width;

export default class Comprar extends React.Component {
    state = {
        showModal: false,
        status: "Pending",
        id: "",
        link: ""
    };
    handleResponse = data => {
        if (data.title === "success") {
            this.setState({ showModal: false, status: "Complete" });
        } else if (data.title === "cancel") {
            this.setState({ showModal: false, status: "Cancelled" });
        } else {
            return;
        }
    };

    componentDidMount = async () =>{
        console.log("-----------PRODUCT-----------");
        console.log(this.props.product.id);
        console.log("-----------USER-----------");
        console.log(this.props.user);
        let URL = `${API_BASE}/paypal/compra_producto/${this.props.product}/${this.props.user}`;
        axios
            .post(URL, {}, {})
            .then(resp => {
                console.log("----------ID-----------");
                console.log(resp.id);
                console.log("----------LINK-----------");
                console.log(resp.link);
                this.setState({ id: resp.id });
                this.setState({ link: resp.link });
            })
            .catch(err => {
                console.log(err);
            });
    };

    render() {
        return (
            <View style={styles.root} style={{ marginTop: 0 }} >
            <RkAvoidKeyboard>
                <View style={styles.header}>
                <Modal
                    visible={this.state.showModal}
                    onRequestClose={() => this.setState({ showModal: false })}
                >
                    <WebView
                        source={{ uri: this.state.link }}
                        onNavigationStateChange={data =>
                            this.handleResponse(data)
                        }
                        injectedJavaScript={`document.f1.submit()`}
                    />
                </Modal>
                <View style={styles.section}>
                    <View style={[styles.heading]} style={{marginTop: 150, marginLeft: 25}}>
                        <TouchableOpacity style={styles.button}
                            onPress={() => this.setState({ showModal: true })}
                        >
                            <RkText rkType="header6 primary" style={{fontSize: 45, marginLeft: 5}}>Pay with Paypal</RkText>
                        </TouchableOpacity>
                    </View>
                </View>
            </View>
            <View style={styles.row} style={{marginLeft: width/3 - 15 , marginTop: 600}}>
                <RkText>Payment Status: {this.state.status}</RkText>
            </View>
            </RkAvoidKeyboard>
            </View>
        );
    }
}


const styles = RkStyleSheet.create(theme => ({
    root: {
      backgroundColor: theme.colors.screen.base
    },
    header: {
      backgroundColor: theme.colors.screen.neutral,
      paddingVertical: 25,
      alignItems: "center",
      justifyContent: "center",
      flex: 1
    },
    section: {
      marginVertical: 15
    },
    heading: {
      paddingBottom: 12.5
    },
    row: {
      flexDirection: "row",
      paddingHorizontal: 17.5,
      borderBottomWidth: StyleSheet.hairlineWidth,
      alignItems: "center"
    },
    button: { 
        backgroundColor: 'white', 
        width: 300, 
        height: 100,
        borderColor: 'black'
    }
  }));