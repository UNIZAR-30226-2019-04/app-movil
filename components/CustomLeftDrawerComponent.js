import React, { Component } from "react";
import { View, StyleSheet, SafeArea, ScrollView } from "react-native";
import { Avatar, Tooltip, Text } from "react-native-elements";
import axios from "axios";
import { API_BASE, DEBUG, USER, TOKEN } from "../config";
import { AsyncStorage } from "react-native";
import { DrawerItems } from "react-navigation";

export default class CustomLeftDrawerComponent extends Component {
  state = {
    profile: {},
    imagen_perfil: "http://34.90.77.95:10080/user/default.png"
  };

  componentDidMount() {
    this.fetchData();
  }
  fetchData = async () => {
    let token, user;
    try {
      user = await AsyncStorage.getItem("user");
      token = await AsyncStorage.getItem("token");
      console.log("User", user, token);
    } catch (error) {
      console.log(error);
    }
    if (DEBUG) {
      user = USER;
      token = TOKEN;
    }
    const URL = `${API_BASE}/user/${user}`;
    const res = await axios.get(URL, {
      headers: {
        "Content-Type": "application/json",
        Authorization: token
      }
    });
    const perfil = res.data;
    console.log("Response Drawer Perfil", perfil);
    this.setState({ profile: perfil, imagen_perfil: perfil.imagen_perfil });
  };

  render() {
    return (
      <View style={{ flex: 1 }}>
        <View
          style={{
            marginTop: 30,
            height: 100,
            backgroundColor: "white",
            alignItems: "center",
            justifyContent: "center"
          }}
        >
          <Avatar
            rounded
            onPress={() => console.log("Works!")}
            size="large"
            containerStyle={{ marginHorizontal: 10 }}
            source={{
              uri: this.state.imagen_perfil
            }}
          />
        </View>

        <ScrollView>
          <DrawerItems {...this.props} />
        </ScrollView>
      </View>
    );
  }
}
