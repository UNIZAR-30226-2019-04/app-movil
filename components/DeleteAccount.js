import React, { Component } from "react";
import {
  View,
  Text,
  StyleSheet,
  Dimensions,
  TouchableOpacity
} from "react-native";
import { Button, ButtonGroup, Header, Input } from "react-native-elements";

import { RkStyleSheet } from "react-native-ui-kitten";
import axios from "axios";
import { API_BASE, API_KEY } from "../config";

const width = Dimensions.get("window").width;
const height = Dimensions.get("window").height;

class DeleteAccount extends Component {
  constructor(props) {
    super(props);
  }

  state = {
    email: "",
    password: "",
    token: "",
    selectedIndex: null,
    isPasswordValid: true,
    errorMsg: ""
  };

  onPasswordInputChanged = text => {
    this.setState({ password: text });
  };

  static navigationOptions = ({ navigation }) => ({
    title: navigation.getParam("receiver"),
    headerTitleStyle: { textAlign: "center", alignSelf: "center" },
    headerStyle: {
      backgroundColor: "white"
    }
  });

  _deleteUser() {
    const { password } = this.state;
    const public_id = this.props.user;
    const token = this.props.token;
    console.log("_deleteUser", public_id, token);

    axios
      .post(
        `${API_BASE}/user/${public_id}/delete`,
        {
          password: password
        },
        {
          headers: {
            "Content-Type": "application/json",
            Authorization: token
          }
        }
      )
      .then(resp => {
        console.log(resp.data);
        console.log("_deleteUser", resp.data);

        if (err.response.status !== 200) {
          this.setState({
            errorMsg: "Contraseña incorrecta.",
            isPasswordValid: false
          });
        }
        this.props.navigation.navigate("Welcome");
        //Aquí la parte de borrar
      })
      .catch(err => {
        console.log("Error: ", err.response.status);
      });
  }

  render() {
    return (
      <View>
        <View style={styles.section}>
          <Header
            centerComponent={{
              text: "BORRAR CUENTA",
              style: {
                color: "#fff",
                fontSize: 24,
                marginBottom: 20
              }
            }}
            backgroundColor="#0864AF"
          />
          <View style={styles.row}>
            <Input
              label="Introduzca su contraseña:"
              value={this.state.password}
              //rkType="right clear"
              secureTextEntry
              onChangeText={this.onPasswordInputChanged}
              errorMessage={
                this.state.isPasswordValid ? null : this.state.errorMsg
              }
            />
          </View>
          <Button
            containerStyle={{ height: 180, marginHorizontal: 30 }}
            style={styles.button}
            title="BORRAR"
            onPress={() => this._deleteUser()}
          />
        </View>
      </View>
    );
  }
}
export default DeleteAccount;

const styles = RkStyleSheet.create(theme => ({
  root: {
    backgroundColor: theme.colors.screen.base
  },
  padding: {
    padding: 20,
    height: 200
  },
  section: {
    backgroundColor: "#F5F5F5",
    height: height
  },
  reportButton: {
    width: width / 2 + 30,
    height: 90,
    borderRadius: 9,
    borderWidth: 4,
    borderColor: "red",
    marginTop: 35,
    marginBottom: 35,
    marginLeft: width / 4 - 15
  },
  row: {
    flexDirection: "row",
    paddingHorizontal: 17.5,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderColor: theme.colors.border.base,
    alignItems: "center",
    marginTop: 50
  },
  button: {
    paddingVertical: 30,
    flex: 1,
    height: 30
  }
}));
