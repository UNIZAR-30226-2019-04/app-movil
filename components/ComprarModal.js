import React, { Component } from "react";
import {
  Modal,
  Text,
  TouchableHighlight,
  View,
  ScrollView,
  StyleSheet,
  Dimensions
} from "react-native";
import { SearchBar } from "react-native-elements";
import { Platform, StatusBar, TouchableOpacity, WebView } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { Button, Tooltip } from "react-native-elements";
import Icon from "react-native-vector-icons/FontAwesome";
import categories from "../assets/categorias.json";
import Comprar from "./Comprar";

const width = Dimensions.get("window").width;
const height = Dimensions.get("window").height;

export default class ComprarModal extends Component {
  constructor(props) {
    super(props);
  }
  state = {
    modalVisible: false,
    status: ""
  };

  handleResponse = data => {
    if (data.title == "success") {
      this.setState({ modalVisible: false, status: "Complete" });
    } else if (data.title == "cancel") {
      this.setState({ modalVisible: false, status: "Cancelled" });
    } else {
      return;
    }
  };

  setModalVisible(visible) {
    this.setState({ modalVisible: visible });
  }
  render = () => {
    if (!this.props.trueque) {
      return (
        <View style={{ marginTop: 0 }}>
          <Modal
            animationType="slide"
            transparent={false}
            visible={this.state.modalVisible}
            onRequestClose={() => this.setModalVisible(false)}
          >
            <Comprar closeModal={() => this.setModalVisible(false)} />

            <Ionicons
              style={{
                position: "absolute",
                right: 0,
                marginRight: 20,
                marginLeft: 20,
                marginTop: 60
              }}
              onPress={() => {
                console.log("Pressed");
                this.setModalVisible(false);
              }}
              name="md-close"
              size={38}
              color="grey"
            />
          </Modal>

          <Button
            title="Comprar"
            titleStyle={{
              margin: 10,
              width: width / 2 - 40
            }}
            buttonStyle={{
              backgroundColor: "mediumseagreen"
            }}
            onPress={() => this.setModalVisible(true)}
          />
        </View>
      );
    } else {
      return null;
    }
  };
}

const styles = StyleSheet.create({
  section: {
    flex: 1,
    marginTop: 52,
    backgroundColor: "white",
    borderRadius: 6,
    borderWidth: 2,
    borderColor: "#fff"
  },

  floatingButton: {
    width: 160,
    height: 60
  },

  lineStyle: {
    height: 1,
    borderWidth: 0.3,
    borderColor: "grey",
    margin: 4,
    marginHorizontal: 24
  }
});
