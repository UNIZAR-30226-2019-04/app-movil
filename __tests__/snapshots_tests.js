import React from 'react';

import Conversaciones from '../screens/Conversaciones';
import Chat from '../screens/Chat';
import DashboardScreen from '../screens/DashboardScreen';
import HomeScreen from '../screens/HomeScreen';
import LinksScreen from '../screens/LinksScreen';
import ProductDetails from '../screens/ProductDetails';
import WelcomeScreen from '../screens/WelcomeScreen';
import CameraRollSelect from '../components/CameraRollSelect';
import CategoryPickerModal from '../components/CategoryPickerModal';
import ComprarModal from '../components/ComprarModal';
import Comprar from '../components/Comprar';
import CustomLeftDrawerComponent from '../components/CustomLeftDrawerComponent';
import DeleteAccount from '../components/DeleteAccount';
import MakeReview from '../components/MakeReview';
import Review from '../components/Review';
import ReviewModal from '../components/ReviewModal';
import SubastarModal from '../components/SubastarModal';
import TypePickerModal from '../components/TypePickerModal';
import UploadMultimedia from '../components/UploadMultimedia';

import renderer from 'react-test-renderer';


describe('Screens snapshot tests', () => {

  test('Conversaciones renders correctly', () => {
    const tree = renderer.create(<Conversaciones />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Chat renders correctly', () => {
    const tree = renderer.create(<Chat />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('DashboardScreen renders correctly', () => {
    const tree = renderer.create(<DashboardScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('HomeScreen renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('LinksScreen renders correctly', () => {
    const tree = renderer.create(<LinksScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ProductDetails renders correctly', () => {
    const tree = renderer.create(<ProductDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('WelcomeScreen renders correctly', () => {
    const tree = renderer.create(<WelcomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Components snapshot tests', () => {

  test('CameraRollSelect renders correctly', () => {
    const tree = renderer.create(<CameraRollSelect />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CategoryPickerModal renders correctly', () => {
    const tree = renderer.create(<CategoryPickerModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Comprar renders correctly', () => {
    const tree = renderer.create(<Comprar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ComprarModal renders correctly', () => {
    const tree = renderer.create(<ComprarModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CustomLeftDrawerComponent renders correctly', () => {
    const tree = renderer.create(<CustomLeftDrawerComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('DeleteAccount renders correctly', () => {
    const tree = renderer.create(<DeleteAccount />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('MakeReview renders correctly', () => {
    const tree = renderer.create(<MakeReview />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Review renders correctly', () => {
    const tree = renderer.create(<Review />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ReviewModal renders correctly', () => {
    const tree = renderer.create(<ReviewModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('SubastarModal renders correctly', () => {
    const tree = renderer.create(<SubastarModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('TypePickerModal renders correctly', () => {
    const tree = renderer.create(<TypePickerModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('UploadMultimedia renders correctly', () => {
    const tree = renderer.create(<UploadMultimedia />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});








