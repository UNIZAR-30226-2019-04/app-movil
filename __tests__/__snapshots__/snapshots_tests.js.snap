// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        },
        Object {
          "height": 0,
        },
        Object {
          "marginVertical": 20,
        },
      ]
    }
  />
</View>
`;

exports[`Conversaciones renders correctly 1`] = `null`;

exports[`DashboardScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    DashboardScreen
  </Text>
</View>
`;

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
/>
`;

exports[`LinksScreen renders correctly 1`] = `null`;

exports[`ProductDetails renders correctly 1`] = `null`;

exports[`WelcomeScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 1334,
        "justifyContent": "center",
        "left": 0,
        "resizeMode": "stretch",
        "top": 0,
        "width": 750,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <Text>
      Cargando...
    </Text>
  </View>
</View>
`;
